planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/pull_over_planner/bezier_pull_over.cpp

```diff
@@ -0,0 +1,406 @@
+    const auto span = static_cast<unsigned>(
+      std::max<int>(static_cast<int>(to_idx) - static_cast<int>(from_idx), 0));
+    const auto reference_curvature =
+      autoware::motion_utils::calcCurvature(processed_prev_module_path->points);
+    const auto & from_pose = shifted_path.path.points[from_idx].point.pose;
+    const auto & to_pose = shifted_path.path.points[to_idx].point.pose;
+    const autoware::sampler_common::State initial{
+      {from_pose.position.x, from_pose.position.y},
+      {0.0, 0.0},
+      reference_curvature.at(from_idx),
+      tf2::getYaw(from_pose.orientation)};
+    const autoware::sampler_common::State final{
+      {to_pose.position.x, to_pose.position.y}, {0.0, 0.0}, 0.0, tf2::getYaw(to_pose.orientation)};
+    // setting the initial velocity to higher gives straight forward path (the steering does not
+    // change)
+    const auto bezier_path =
+      bezier_sampler::sample(initial, final, v_init_coeff, v_final_coeff, acc_coeff);
+    const auto bezier_points = bezier_path.cartesianWithHeading(span);
+    for (unsigned i = 0; i + 1 < span; ++i) {
+      auto & p = shifted_path.path.points[from_idx + i];
+      p.point.pose.position.x = bezier_points[i].x();
+      p.point.pose.position.y = bezier_points[i].y();
+      p.point.pose.orientation =
+        universe_utils::createQuaternionFromRPY(0.0, 0.0, bezier_points[i].z());
+    }
+    shifted_path.path.points =
+      autoware::motion_utils::removeOverlapPoints(shifted_path.path.points);
+
+    autoware::motion_utils::insertOrientation(shifted_path.path.points, true);
+
+    // set goal pose with velocity 0
+    {
+      PathPointWithLaneId p{};
+      p.point.longitudinal_velocity_mps = 0.0;
+      p.point.pose = goal_pose;
+      p.lane_ids = shifted_path.path.points.back().lane_ids;
+      for (const auto & lane : shoulder_lanes) {
+        p.lane_ids.push_back(lane.id());
+      }
+      shifted_path.path.points.push_back(p);
+    }
+
+    // set lane_id and velocity to shifted_path
+    for (size_t i = path_shifter.getShiftLines().front().start_idx;
+         i < shifted_path.path.points.size() - 1; ++i) {
+      auto & point = shifted_path.path.points.at(i);
+      // set velocity
+      point.point.longitudinal_velocity_mps =
+        std::min(point.point.longitudinal_velocity_mps, static_cast<float>(pull_over_velocity));
+
+      // add target lanes to points after shift start
+      // add road lane_ids if not found
+      for (const auto id : shifted_path.path.points.back().lane_ids) {
+        if (std::find(point.lane_ids.begin(), point.lane_ids.end(), id) == point.lane_ids.end()) {
+          point.lane_ids.push_back(id);
+        }
+      }
+      // add shoulder lane_id if not found
+      for (const auto & lane : shoulder_lanes) {
+        if (
+          std::find(point.lane_ids.begin(), point.lane_ids.end(), lane.id()) ==
+          point.lane_ids.end()) {
+          point.lane_ids.push_back(lane.id());
+        }
+      }
+    }
```

Comment:
better not to set too many id.
you can copy the code from shift pull over.
refer: https://github.com/autowarefoundation/autoware.universe/pull/9360

Reply:
fixed in 96f632d1b17f2702d987d14e1ff7be5b65be0842